src/internal/connector/onedrive/item.go

```diff
@@ -265,10 +266,28 @@ func filterUserPermissions(perms []models.Permissionable) []UserPermission {
 			continue
 		}
 
+		entityID := ""
+		if gv2.GetUser() != nil {
+			entityID = *gv2.GetUser().GetId()
+		} else if gv2.GetGroup() != nil {
+			entityID = *gv2.GetGroup().GetId()
+		} else if gv2.GetApplication() != nil {
+			// TODO(meain): Applications permissions restore not
+			// tested manually
+			entityID = *gv2.GetApplication().GetId()
+		}
+
+		// Technically GrantedToV2 can also contain devices, but the
+		// documentation does not mention about devices in permissions
+		if entityID == "" {
+			// This should ideally not be hit
+			continue
+		}
+
 		up = append(up, UserPermission{
 			ID:         ptr.Val(p.GetId()),
 			Roles:      roles,
-			Email:      *p.GetGrantedToV2().GetUser().GetAdditionalData()["email"].(*string),
+			EntityID:   entityID,
```

Comment:
is overloading this field alright for restores? For example, does Graph API need to know if we're sending it a user ID or a group ID?

Reply:
Yup, I am able to restore both users and groups with just id. The [docs](https://learn.microsoft.com/en-us/graph/api/resources/driverecipient?view=graph-rest-1.0) also mention the field in restore as `objectId` and nothing specific.